joint_state_broadcaster/doc/userdoc.rst

```diff
@@ -13,13 +13,32 @@ Broadcasters are not real controllers, and therefore take no commands.
 Hardware interface type
 -----------------------
 
-All available *joint state interfaces* are used by this broadcaster.
+All available *joint state interfaces* are used, or interfaces defined by the ``joints`` and ``interfaces`` parameters of both are not empty.
+In the latter case, resulting interfaces is a "matrix" of interfaces resulting from the cross-product of those two vectors.
+If some requested interfaces are missing, the controller will print a warning about that, but work for other interfaces.
+If none of the requested interface are not defined, the controller returns error on activation.
```

Comment:
```suggestion
By default, all available *joint state interfaces* are used, unless configured otherwise.
In the latter case, resulting interfaces is defined by a "matrix" of interfaces resulting from the cross-product of the ``joints`` and ``interfaces`` parameters.
If some requested interfaces are missing, the controller will print a warning about that, but work for other interfaces.
If none of the requested interface are not defined, the controller returns error on activation.
```